src/main/java/net/rubygrapefruit/platform/internal/jni/DefaultOsxFileEventFunctions.java

```diff
@@ -0,0 +1,73 @@
+package net.rubygrapefruit.platform.internal.jni;
+
+import net.rubygrapefruit.platform.NativeException;
+import net.rubygrapefruit.platform.NativeIntegration;
+import net.rubygrapefruit.platform.internal.FunctionResult;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Queue;
+import java.util.Set;
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+public class DefaultOsxFileEventFunctions implements NativeIntegration {
+
+    public boolean addRecursiveWatch(String path) {
+        FunctionResult result = new FunctionResult();
+        OsxFileEventFunctions.createWatch(path, result);
+        if (result.isFailed()) {
+            throw new NativeException("Failed to start watching " + path + ". Reason: " + result.getMessage());
+        }
+        return true;
+    }
+
+    interface ChangeCallback {
+        @SuppressWarnings("unused")
+        // invoked from native code
+        void pathChanged(String path);
+    }
+
+    public static class ChangeCollector implements ChangeCallback {
+        private Queue<String> globalQueue = new ConcurrentLinkedQueue<String>();
+
+        public void pathChanged(String path) {
+            System.out.println("> Changed: " + path);
+            globalQueue.add(path);
+        }
+
+        private List<String> getAllChanges() {
+            Set<String> seen =  new HashSet<String>((int) (globalQueue.size() / 0.75f) + 1, 0.75f);
+            List<String> result = new ArrayList<String>(globalQueue.size());
+            while (true) {
+                String path = globalQueue.poll();
+                if (path == null) {
+                    break;
+                }
+                if (seen.add(path)) {
+                    result.add(path);
+                }
+            }
+            return result;
+        }
+    }
+
+    public ChangeCollector startWatch() {
+        ChangeCollector collector = new ChangeCollector();
+        FunctionResult result = new FunctionResult();
+        OsxFileEventFunctions.startWatch(collector, result);
+        if (result.isFailed()) {
+            throw new NativeException("Failed to start collecting changes. Reason: " + result.getMessage());
+        }
+        return collector;
+    }
+
+    public List<String> stopWatch(ChangeCollector collector) {
```

Comment:
There's so much static state in the implementation that it seems odd that the collector is a parameter here, rather than just more static state.

Same for `startWatch()` and its return value.